web/auth/auth.go

```diff
@@ -530,4 +469,8 @@ func Routes(router *echo.Group) {
 
 	router.POST("/access_token", accessToken)
 	router.POST("/secret_exchange", secretExchange)
+
+	// 2FA
+	router.GET("/twofactor", twoFactorForm, noCSRF)
+	router.POST("/twofactor", twoFactor, noCSRF)
```

Comment:
What is the rationale for using the `noCSRF` middleware?

Reply:
It is a mistake, nice pick !